draft-cms-masque-ip-proxy-reqs.md

```diff
@@ -133,30 +143,33 @@ packets, in their unmodified entirety. The protocol will support both IPv6
 
 ## Maximum Transmission Unit
 
-The protocol will allow endpoints to negotiate the Maximum Transmission Unit
-(MTU) in use over a given Data Transport. This will allow avoiding IP
-fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
+The protocol will allow endpoints to inform each other of the Maximum
+Transmission Unit (MTU) they are willing to forward. This will allow avoiding
+IP fragmentation, especially as IPv6 does not allow IP fragmentation by nodes
 along the path.
 
 ## IP Assignment
 
-Both the client or server may request to be assigned an IP address range. In
-response to the request, the peer will respond with an IP address range of its
-choosing.
+The server will be able to assign an IP address of its choosing to the client.
+The client may suggest a preference of IP address which the server can choose
+to follow. Similarly, to support the network-to-network use case, the client
```

Comment:
For the network-to-network use case, I had been expecting that we'd use the route negotiation approach -- with either side saying "here's the set of IP ranges that you can talk to in my network". This is assuming that both sides already have an address space that they own and they're just looking to expose it to the other side.

Are you thinking of some other scenario?

Reply:
I was thinking of the case where the client owns an intranet prefix and can tell the server "here's an IP address for you from my intranet"

Reply:
Hm, but in that scenario, how do we know that the server _wants_ to have an address within the client's intranet? Or are you saying that the client handing the server an ip address range is the client's way of telling the server that it should use that address?

Reply:
Makes sense, I reverted this change and made address assignment request-response again.